jsonparse.js

```diff
@@ -267,7 +267,7 @@ proto.write = function (buffer) {
               return this.charError(buffer, i);
             }
 
-            if ((this.string.match(/[0-9]+/) == this.string) && (result.toString() != this.string)) {
+            if ((this.string.match(/-?[0-9]+/) == this.string) && (result.toString() != this.string)) {
```

Comment:
Is there a reason that this does a match + compare rather than just a test with `^/-?[0-9]+/$`, seems like it might be a bit more efficient

Reply:
I don't know, why it the weird match + compare use there, it was not written by me. I agree with you, it will be better to use just a test.